tests/internal/CMakeLists.txt

```diff
@@ -269,6 +269,132 @@ if(ENABLE_HIP)
   add_subdirectory(src/hip_defines_test)
 endif()
 
+#------------------------------------------------------------------------------
```

Comment:
👍 
This is really nice! Thanks for adding it.

I wonder if it should be pulled out into its own file and included here instead of being inlined?

Reply:
we need a much larger followup PR that increases testing of BLT .. i can pull it into its own file